armsrc/iso14443a.c

```diff
@@ -2549,12 +2551,14 @@ void RAMFUNC SniffMifare(uint8_t param) {
 					DemodReset();
 				}
 				ReaderIsActive = (Uart.state != STATE_UNSYNCD);
+				TagIsActive = !ReaderIsActive;
```

Comment:
You assume that reader and tag are sending one after the other. This is not necessarily the case. The case that `TagIsActive` and `ReaderIsActive` are both `false` must not be excluded.